iap/spec/iap_spec.rb

```diff
@@ -17,30 +17,31 @@
 require "stringio"
 
 def fetch_iap_jwt
-  begin
-    prev = $stdout
-    $stdout = StringIO.new
-    make_iap_request url: "https://print-iap-jwt-assertion-dot-cloud-iap-for-testing.uc.r.appspot.com",
-      client_id: "1031437410300-ki5srmdg37qc6cl521dlqcmt4gbjufn5.apps.googleusercontent.com"
-    result = $stdout.string
-  ensure
-    $stdout = prev
+  url = "https://print-iap-jwt-assertion-dot-cloud-iap-for-testing.uc.r.appspot.com"
+  client_id = "1031437410300-ki5srmdg37qc6cl521dlqcmt4gbjufn5.apps.googleusercontent.com"
+  resp = make_iap_request url: url, client_id: client_id
+  tries = 3
+  while resp.status != 200 && tries > 0
+    tries -= 1
+    sleep 5
```

Comment:
Consider only sleeping when the status `!= 200`, that way the happy path is fast.